plasmapy/utils/exceptions.py

```diff
@@ -93,6 +94,14 @@ class PlasmaPyWarning(Warning):
     pass
 
 
+class PlasmaPyDeprecationWarning(DeprecationWarning, PlasmaPyWarning):
+    """
+    A warning for features in PlasmaPy that have been deprecated.
+    """
+
+    pass
+
+
```

Comment:
What's the motivation for this, versus just using `DeprecationWarning`?  If we really want our own `DeprecationWarning`, then I would encourage considering an approach similar to how `sphinx` does it.   They do this [this](https://github.com/sphinx-doc/sphinx/blob/64498ad0dbab86fa42f5556d7bc84baf719cefda/sphinx/deprecation.py#L21-L33)...

```python
class RemovedInSphinx40Warning(DeprecationWarning):
    pass


class RemovedInSphinx50Warning(PendingDeprecationWarning):
    pass


class RemovedInSphinx60Warning(PendingDeprecationWarning):
    pass
```

So, our version would look like...

```python
class RemovedInPlasmaPy05Warning(DeprecationWarning, PlasmaPyWarning):
    pass


class RemovedInPlasmaPy06Warning(PendingDeprecationWarning, PlasmaPyWarning):
    pass
```

Now, I also think we should consider `FutureWarning` versus `DeprecationWarning`.  According to [Python's documentation](https://docs.python.org/3/library/warnings.html?highlight=warning#warning-categories), `DeprecationWarning` and `FutureWarnings` both announce deprecated features, but `DeprecationWarning` is intended for developers and `FutureWarning` is intended for end users.  Since we are constantly changing and deprecating end user features, we might want to consider using `FutureWarning` instead.

Reply:
My original motivation was to follow Astropy's strategy for handling deprecations.  The benefit of having `PlasmaPyDeprecationWarning` is to have a bit more specificity than `DeprecationWarning`.  I'm more inclined to specify the version in which something is deprecated in a `@deprecated` decorator (#945) rather than having a series of different classes.  I don't have a strong opinion though.

I think I'll back off on the deprecation stuff in this PR since it's off-topic, but I'll link to this discussion in #945 since it's really relevant there.